lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -329,32 +330,28 @@ def run_sync(command, stdin = nil)
       stdout = ''
       stderr = ''
       exit_status = nil
-      started = false
 
-      channel = session.open_channel do |ch|
-        ch.on_data do |c, data|
-          stdout.concat(data)
-        end
-        ch.on_extended_data { |_, _, data| stderr.concat(data) }
-        ch.on_request('exit-status') { |_, data| exit_status = data.read_long }
-        # Send data to stdin if we have some
-        ch.send_data(stdin) unless stdin.nil?
-        # on signal: sending the signal value (such as 'TERM')
-        ch.on_request('exit-signal') do |_, data|
-          exit_status = data.read_string
-          ch.close
-          ch.wait
-        end
-        ch.exec command do |_, success|
-          raise 'could not execute command' unless success
-
-          started = true
+      master, slave = PTY.open
+      read, write = IO.pipe
+      pid = spawn('/usr/bin/ssh', @ssh_user + '@' + @host, command, :in => read, :out => slave)
+      read.close
+      slave.close
+
+      unless stdin.nil?
```

Comment:
Do I read it right that if there's no input, it does not read the output of the command being run?

Reply:
Yes, because we checked if input is not empty when we used Net::SSH module:
```ruby
channel.send_data(stdin) unless stdin.nil?
```

Reply:
Well yes, but originally if the input was empty, we didn't send it over but we still read the output. Currently it seems that if there's no input, we discard all the output that the remote command produces

Reply:
Oh okay, I understand now, thanks for the notice :+1: